compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala

```diff
@@ -267,6 +267,23 @@ trait PrepJSExports[G <: Global with Singleton] { this: PrepJSInterop[G] =>
         Nil
     }
 
+    val hasDuplicateExportAnnotation = directAnnots.count { annot =>
+      annot.symbol == JSExportAnnotation || annot.symbol == JSExportNamedAnnotation
+    } > 1
+    if (hasDuplicateExportAnnotation) {
+      reporter.warning(sym.pos,
+        s"Found duplicate @JSExport")
+    }
```

Comment:
Hum, wouldn't this warn for this:

```scala
@JSExport("a")
@JSExport("b")
def foo(): Int = ...
```

Reply:
Do you mean such case should not be earned?

Reply:
Yes, exactly. It is reasonable to export a method under multiple names.

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/3633/commits/6a048ef99cb6c7df95cb08e809b86ffce474feb8

Reply:
Now from what I understand this wouldn't warn on:

```scala
@JSExport("foo")
@JSExport
def foo(): Unit = ...
```

Have you considered using the post-processed `ExportInfo`s? They neatly give you export type, destination and name for grouping.

Something like:

```scala
allExportInfos.groupBy(i => (i.jsName, i.Destination, i.isNamed).filter(_.size > 1).foreach {
  reporter.warning(...)
}
```

Reply:
> 
> 
> Now from what I understand this wouldn't warn on:
> 
> ```scala
> @JSExport("foo")
> @JSExport
> def foo(): Unit = ...
> ```

I think that case is covered in https://github.com/scala-js/scala-js/pull/3633/files#diff-c700423ea542bd0634232c51084a9acaR47




> Have you considered using the post-processed `ExportInfo`s? They neatly give you export type, destination and name for grouping.

I will try refactoring using that.

Reply:
You could still report at `sym.pos` instead of `i.pos`, no? I haven't seen how it looks, but I expect it to be much less duplicated logic if you use `exportInfo`.

Reply:
Hmm, I tried refactoring using `allExportInfos` but I could not since [lack of IDE support](https://github.com/scala-js/scala-js/issues/3638) give a hurdle for me to understand code base and modify...
I have a little time for cosmetic change, but not enough for refactoring since I will be busy for several weeks.
Could you consider review PR as-is and accept refactoring in future PR? 
Or, you can push changes to the branch since the branch allow edits from maintainers.

Reply:
> Could you consider review PR as-is and accept refactoring in future PR?

I'd rather not. We have a long history on trying to accumulate a minimum of technical debt. (Best example probably being #3240).

> Or, you can refactor and push changes to the branch since the branch allow edits from maintainers.

I'll do that if I have a moment. Otherwise you can take it again. "several weeks" is in my experience not a very long time, especially if you're only contributing in your free time (like I do).

Reply:
I found time to refactor using `allExportInfos` !!